packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,143 @@
+/// tree to remain marked not sensitive as long as there are (1) no other `SensitiveContent`
+/// widgets in the tree or (2) there  are only other not sensitive `SensitiveContent` widgets
+/// in the tree. If there are no `SensitiveContent` widgets in the tree, the default setting as
+/// queried from the embedding will be used. This could be set by a Flutter developer in native
+/// Android; otherwise, Android uses [ContentSensitivity.autoSensitive] by default.
+/// {@endtemplate}
+///
+/// * See [SensitiveContent] for how to set a [ContentSensitivity] level
+///   in order for sensitive content to be obscured when the Flutter screen
+///   is shared.
+enum ContentSensitivity {
+  /// Content sensitivity is auto-detected by the native platform.
+  ///
+  /// When this level is set via a [SensitiveContent] widget, the window
+  /// hosting the screen will only be marked as sensitive if other [SensitiveContent]
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_AUTO for how
+  /// this mode behaves on native Android.
+  ///
+  /// This mode currently does not work for Flutter, and thus, will never obscure the screen during
+  /// an active media projection. However, it will still logically be prioritized over [notSensitive].
+  // TODO(camsim99): Implement `autoSensitive` mode that matches the behavior
+  // of `CONTENT_SENSITIVITY_AUTO` on Android that has implemented based on autofill hints; see
+  // https://github.com/flutter/flutter/issues/160879.
+  autoSensitive(id: 0),
+
+  /// The widget tree contains sensitive content.
+  ///
+  /// When this level is set via a [SensitiveContent] widget, the window
+  /// hosting the screen will be marked as sensitive during an active media
+  /// projection session.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+  sensitive(id: 1),
+
+  /// The widget tree does not contain sensitive content.
+  ///
+  /// When this level is set via a [SensitiveContent] widget, the window
+  /// hosting the screen will only be marked as sensitive if other [SensitiveContent]
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+  notSensitive(id: 2);
+
+  const ContentSensitivity({required this.id});
+
+  /// Identifier for each [ContentSensitivity] level, which matches the native Android value
+  /// of the constant that each level corresponds to.
+  ///
+  /// See https://developer.android.com/reference/android/view/View for the enum values for
+  /// the corresponding Android constants (CONTENT_SENSITIVITY_AUTO, CONTENT_SENSITIVITY_SENSITIVE,
+  /// CONTENT_SENSITIVITY_NOT_SENSITIVE).
+  final int id;
```

Comment:
it feels weird that we define the dart enum to platform enum translation in the framework. I think they should be done in the platform embedding. Imagine in the future we may need to add another platform for this feature and it uses a different number for each of the const in Android.

I think here we should just use a plain enum, and send the value to android directly. It will then be translated into Android constants there. Same thing when android sends back the setting.

Reply:
> I think here we should just use a plain enum, and send the value to android directly. It will then be translated into Android constants there. Same thing when android sends back the setting.

What do you mean by this? Wouldn't this also be clunky because we can't send an enum right? So it would have to be like a String to Android enum translation on the platform side?

Reply:
I guess I could see doing this if we may need to add another platform. Will do

Reply:
> Wouldn't this also be clunky because we can't send an enum right? So it would have to be like a String to Android enum translation on the platform side?

enum will send as an int, and the android needs to create a lookup from that int to its own constant

Reply:
I used the enum name instead, but if anyone has strong opinions, let me know